platform/security/pdp/security-pdp-authzrealm/src/main/java/ddf/security/pdp/realm/SimpleAuthzRealm.java

```diff
@@ -280,61 +278,69 @@ private boolean isPermitted(PrincipalCollection subjectPrincipal, Permission per
                     }
                 }
 
-                CollectionPermission subjectAllCollection = new CollectionPermission(perms);
-                KeyValueCollectionPermission matchAllCollection = new KeyValueCollectionPermission(matchAllPermissions);
-                KeyValueCollectionPermission matchOneCollection = new KeyValueCollectionPermission(matchOnePermissions);
+                CollectionPermission subjectAllCollection = new CollectionPermission(
+                        CollectionPermission.NO_ACTION, perms);
+                KeyValueCollectionPermission matchAllCollection = new KeyValueCollectionPermission(
```

Comment:
The matchAll and matchOne permissions shouldn't be set to NO_ACTION, they should be using whatever action was supplied in the original KeyValueCollectionPermission that got ripped apart above. Otherwise, we'd have no way to do something specific on an action with the policy extensions.